sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java

```diff
@@ -300,23 +310,96 @@ public void stop() {
             });
     }
 
+    @Override
+    public KubernetesClusterImpl update() {
+        parameterSnapshotOnUpdate = this.deepCopyInner();
+        parameterSnapshotOnUpdate.withServicePrincipalProfile(null);    // servicePrincipalProfile is null in update
+        return super.update();
+    }
+
+    boolean isClusterModifiedDuringUpdate(ManagedClusterInner parameter) {
```

Comment:
Yes, current this is such.

Maybe it will be more consistent if modify is also done on agent pool APIs. But that might incur e.g. multiple API calls, if one wanted to update tags on multiple agent pool.
So for now, still keep current behavior.